src/deps.mli

```diff
@@ -0,0 +1,23 @@
+open! Import
+
+type t
+
+val empty : t
+
+val path_diff : t -> t -> Path.Set.t
+
+val path_union : t -> t -> Path.Set.t
+
+val union : t -> t -> t
+
+val trace : t -> Env.t -> (string * string) list
```

Comment:
This needs a comment as it's not obvious what it means. It is basically a value that is guaranteed to change if the set of dependencies denoted by `t` changes, modulo hash collisions.